src/pages/label-template/Forms/LabelTemplateForm.js

```diff
@@ -0,0 +1,180 @@
+// ** MUI Imports
+import { Grid } from '@mui/material'
+
+// ** Custom Imports
+import CustomTextField from 'src/components/Inputs/CustomTextField'
+import ResourceComboBox from 'src/components/Shared/ResourceComboBox'
+import FormShell from 'src/components/Shared/FormShell'
+import { ResourceIds } from 'src/resources/ResourceIds'
+import { RequestsContext } from 'src/providers/RequestsContext'
+import { useContext, useEffect } from 'react'
+import * as yup from 'yup'
+import toast from 'react-hot-toast'
+import { useInvalidate } from 'src/hooks/resource'
+import { useForm } from 'src/hooks/form'
+import { ControlContext } from 'src/providers/ControlContext'
+import CustomNumberField from 'src/components/Inputs/CustomNumberField'
+import { DataSets } from 'src/resources/DataSets'
+import { SCRepository } from 'src/repositories/SCRepository'
+
+const LabelTemplateForm = ({ labels, maxAccess, store, setStore, editMode }) => {
+  const { getRequest, postRequest } = useContext(RequestsContext)
+  const { platformLabels } = useContext(ControlContext)
+  const { recordId } = store
+
+  const invalidate = useInvalidate({
+    endpointId: SCRepository.LabelTemplate.qry
+  })
+
+  const { formik } = useForm({
+    maxAccess,
+    initialValues: {
+      recordId: recordId || null,
+      name: '',
+      width: '',
+      height: '',
+      labelHomeX: '',
+      labelHomeY: '',
+      format: ''
+    },
+    enableReinitialize: false,
+    validateOnChange: false,
+    validationSchema: yup.object({
+      name: yup.string().required(),
+      width: yup.string().required(),
+      height: yup.string().required(),
+      labelHomeX: yup.number().nullable().min(0, ' '),
+      labelHomeY: yup.number().nullable().min(0, ' ')
+    }),
+    onSubmit: values => {
+      postLabelTemplate(values)
+    }
+  })
+
+  const postLabelTemplate = async obj => {
+    await postRequest({
+      extension: SCRepository.LabelTemplate.set,
+      record: JSON.stringify(obj)
+    })
+      .then(res => {
+        if (!obj.recordId) {
+          formik.setFieldValue('recordId', res.recordId)
+          toast.success(platformLabels.Added)
+        } else toast.success(platformLabels.Edited)
+        setStore(prevStore => ({
+          ...prevStore,
+          recordId: res.recordId
+        }))
+        invalidate()
+      })
+      .catch(error => {})
+  }
+
+  useEffect(() => {
+    ;(async function () {
+      try {
+        if (recordId) {
+          const res = await getRequest({
+            extension: SCRepository.LabelTemplate.get,
+            parameters: `_recordId=${recordId}`
+          })
+          var result = res.record
```

Comment:
no need for this var
set res.record to formik directly

Reply:
no because i used res.record more than one